classes/article/PublishedArticleDAO.inc.php

```diff
@@ -105,10 +105,13 @@ function getPublishedArticles($issueId) {
 				' . $this->getFetchColumns() . '
 			FROM	published_submissions ps
 				LEFT JOIN submissions s ON ps.submission_id = s.submission_id
+				JOIN submission_settings ss ON (ss.submission_id = s.submission_id)
 				' . $this->getFetchJoins() . '
 				LEFT JOIN custom_section_orders o ON (s.section_id = o.section_id AND o.issue_id = ?)
 			WHERE	ps.submission_id = s.submission_id
 				AND ps.issue_id = ?
+				AND ss.setting_name = \'datePublished\'
```

Comment:
I don't think the setting_value can be empty, because the publication date can not be removed once it has been set (only changed to another date).